python/paddle/fluid/transpiler/distribute_transpiler.py

```diff
@@ -1116,7 +1142,35 @@ def _is_splited_grad_var(self, var, var_dict):
                     break
         return grad_block
 
-    def _append_pserver_non_opt_ops(self, optimize_block, opt_op, endpoint):
+    def _clone_op(self, program, block, op):
+        inputs = self._get_input_map_from_op(
+            self.origin_program.global_block().vars, op)
+        for key, varlist in inputs.iteritems():
+            if not isinstance(varlist, list):
+                varlist = [varlist]
+            for var in varlist:
+                if var not in program.global_block().vars:
+                    print("Find not in var:\n")
+                    print(var)
+                    print(block.vars)
+                    block.clone_variable(var)
+
+        outputs = self._get_output_map_from_op(
+            self.origin_program.global_block().vars, op)
+        for key, varlist in outputs.iteritems():
+            if not isinstance(varlist, list):
+                varlist = [varlist]
+            for var in varlist:
+                if var not in program.global_block().vars:
+                    print("Find not in var:\n")
+                    print(var)
+                    print(block.vars)
+                    block.clone_variable(var)
+
+        program.global_block().append_op(
+            type=op.type, inputs=inputs, outputs=outputs, attrs=op.attrs)
```

Comment:
why are the ops appended to global_block instead of ```block```? I guess these ops belong to the decay sub_block?

Reply:
It seems all lr_decay_ops belong to global block now, but maybe the block will be nested in the lr_decay_op in the future, so I will put it in the block in next commit, thanks for reviewing